situp-core/src/main/java/com/amazon/situp/pipeline/common/PipelineThreadPoolExecutor.java

```diff
@@ -0,0 +1,82 @@
+package com.amazon.situp.pipeline.common;
+
+import com.amazon.situp.pipeline.Pipeline;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.CancellationException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+public class PipelineThreadPoolExecutor extends ThreadPoolExecutor {
+    private static final Logger LOG = LoggerFactory.getLogger(PipelineThreadPoolExecutor.class);
+    private final Pipeline pipeline;
+
+    public PipelineThreadPoolExecutor(
+            final int corePoolSize,
+            final int maximumPoolSize,
+            final long keepAliveTime,
+            final TimeUnit unit,
+            final BlockingQueue<Runnable> workQueue,
+            final ThreadFactory threadFactory,
+            final Pipeline pipeline) {
```

Comment:
nit: In the processworker you are dependent on lot other members of the pipeline, here you don't have any thing except pipeline::Name, can you not accept a string.